Contest/Assignment/src/org/togetherjava/event/elevator/elevators/ElevatorSystem.java

```diff
@@ -39,11 +53,57 @@ public void requestElevator(int atFloor, TravelDirection desiredTravelDirection)
         //  The human can then enter the elevator and request their actual destination within the elevator.
         //  Ideally this has to select the best elevator among all which can reduce the time
         //  for the human spending waiting (either in corridor or in the elevator itself).
-        System.out.println("Request for elevator received");
+
+        if (elevators.isEmpty()){
+            return;
+        }
+        int calculateDistance = 0;
+
+        Elevator chosenElevator = this.elevators.get(0);
+        if (chosenElevator.getCurrentFloor() < atFloor){
+            calculateDistance = atFloor - chosenElevator.getCurrentFloor();
+        } else if (chosenElevator.getCurrentFloor() > atFloor){
+            calculateDistance = chosenElevator.getCurrentFloor() - atFloor;
+        }
+
+        for (Elevator elevator : elevators) {
+
+            int distance = 0;
+            if (elevator.getCurrentFloor() < atFloor){
+                distance = atFloor - elevator.getCurrentFloor();
+            } else if (elevator.getCurrentFloor() > atFloor){
+                distance = elevator.getCurrentFloor() - atFloor;
+            }
+
+            if (calculateDistance > distance) {
+                calculateDistance = distance;
+                chosenElevator = elevator;
+            }
```

Comment:
logic duplication. could be rewritten to pick it in a single loop without that extra logic beforehand